### PR TITLE
Upload / deletion of articles for teacher dashboard

### PR DESCRIPTION
**Pull request by mbrodt**
This pull request adds 2 new endpoints to the API:
1. An "upload file" endpoint which takes some information about an article, and creates it in the database. This function also creates a map between the cohort where the article was added, and the article id (this is to only display articles for a given class on a class page). 

2. An endpoint to delete previously uploaded articles if the person attempting deletion has access to the cohort associated with the article being deleted.

### CODE REVIEW

File: zeeguu_api/api/teacher_dashboard.py [hunk 1]
```diff
@@ -432,3 +436,81 @@ def update_cohort(cohort_id):
...
+@api.route("/upload_files/<cohort_id>", methods=["POST"])
+@with_session
+def upload_file(cohort_id):
+    '''
+        uploads a file for a cohort with input from a POST request
+    '''
+    if (not has_permission_for_cohort(cohort_id)):
+        flask.abort(401)
+    try:
+        for file in json.loads(request.data):
```
Comment (mircealungu) [inline zeeguu_api/api/teacher_dashboard.py]: file used to be a builtin... also, technically you're not getting a file but rather an `article_data` it seems to me

File: zeeguu_api/api/teacher_dashboard.py [hunk 2]
```diff
@@ -432,3 +436,81 @@ def update_cohort(cohort_id):
...
+            url = Url('teacher-dash-file/{}'.format(datetime.now()))
```
Comment (mircealungu) [inline zeeguu_api/api/teacher_dashboard.py]: I propose `Url('user-article-{}'.format(...)`. what do you think?

Comment (mircealungu) [reply zeeguu_api/api/teacher_dashboard.py]: also, that will have spaces in the name... 
e.g. 

    'user-article-2019-04-02 14:38:07.416790'

is this what you indended?

Comment (mircealungu) [reply zeeguu_api/api/teacher_dashboard.py]: maybe something like this?

    'userarticle/{}'.format(uuid.uuid4().hex)

File: zeeguu_api/api/teacher_dashboard.py [hunk 3]
```diff
@@ -432,3 +436,81 @@ def update_cohort(cohort_id):
...
+                content,
+                summary,
+                published_time,
+                None,  # rss feed
+                language
+            )
+            
+            db.session.add(new_article)
+            db.session.flush()
+            db.session.refresh(new_article)
```
Comment (mircealungu) [inline zeeguu_api/api/teacher_dashboard.py]: this is interesting. 
why do add+flush+refresh?

File: zeeguu_api/api/teacher_dashboard.py [hunk 4]
```diff
@@ -432,3 +436,81 @@ def update_cohort(cohort_id):
...
+
+
+@api.route("/cohort_files/<cohort_id>", methods=["GET"])
+@with_session
+def cohort_files(cohort_id):
+    '''
+        Gets the files associated with a cohort
+    '''
+    cohort = Cohort.find(cohort_id) 
+    articles = CohortArticleMap.get_articles_for_cohort(cohort)
```
Comment (mircealungu) [inline zeeguu_api/api/teacher_dashboard.py]: make sure to refactor this if you rename the function in core!

File: zeeguu_api/api/teacher_dashboard.py [hunk 5]
```diff
@@ -432,3 +436,81 @@ def update_cohort(cohort_id):
...
+    return json.dumps(articles)
+
+@api.route("/delete_article/<cohort_id>/<article_id>", methods=["POST"])
+@with_session
+def delete_article(cohort_id, article_id):
+		'''
```
Comment (mircealungu) [inline zeeguu_api/api/teacher_dashboard.py]: in the GH diff UI there's a lot of indentation in this method.
can you check why? 

Screen Shot 2019-04-02 at 2 47 20 PM

Comment (mbrodt) [reply zeeguu_api/api/teacher_dashboard.py]: I tried to fix this, but in the code it looks normal. Let me know if its still happening

**Comment by mircealungu** [comment]
I see tests are passing, and you clarified my last question. I'm merging this into master.

**Pull request merged**